### ISSUE TITLE
find_windows_sdk_bin_dir fails with partial SDK installs

### ISSUE THREAD

**Issue by iliekturtles** [nabijaczleweli/rust-embed-resource#2]
I get the following error when attempting to compile a crate that depends on rust-embed-resource:

`thread 'main' panicked at 'Are you sure you have RC.EXE in your $PATH?: Error { repr: Os { code: 2, message: "The system cannot find the file specified." } }'`

Reviewing the code `compile_resource` is checking if `rc.exe` exists for the detected SDK. In my case I have a partial install for the Windows Kits 10 that only has `bin/<arch>/ucrt` folders. Windows Kits 8.1 is also installed and has `rc.exe`. If the check for the existence of `rc.exe` is pushed into `find_windows_sdk_bin_dir` then partial installs would fail and allow for older SDKs to be checked before falling back to using whatever `rc.exe` is on the user's `$PATH`.

cc @retep998

**Comment by retep998** [comment]
Ideally this should use the same logic that `rustc` and the `gcc` crate use.

**Comment by nabijaczleweli** [comment]
I think this patch would fix it, right?

```diff
diff --git a/src/windows_msvc.rs b/src/windows_msvc.rs
index 55f7a3a..5e9cfe6 100644
--- a/src/windows_msvc.rs
+++ b/src/windows_msvc.rs
@@ -13,13 +13,8 @@ enum Arch {
 }
 
 pub fn compile_resource(out_dir: &str, prefix: &str, resource: &str) {
-    let rc_path = find_windows_sdk_bin_dir().and_then(|mut x| {
-        x.push("rc.exe");
-        if x.exists() { Some(x) } else { None }
-    });
-
     // `.res`es are linkable under MSVC as well as normal libraries.
-    Command::new(rc_path.as_ref().map_or(Path::new("rc.exe"), Path::new))
+    Command::new(find_windows_sdk_bin_dir().as_ref().map_or(Path::new("rc.exe"), Path::new))
         .args(&["/fo", &format!("{}/{}.lib", out_dir, prefix), resource])
         .status()
         .expect("Are you sure you have RC.EXE in your $PATH?");
@@ -45,6 +40,7 @@ fn find_windows_kits_bin_dir(key: &str, arch: Arch) -> Option<PathBuf> {
         .and_then(|reg_key| reg_key.get_value::<String, _>(key))
         .ok()
         .and_then(|root_dir| try_bin_dir(root_dir, "bin/x86", "bin/x64", arch))
+        .and_then(try_rc_exe)
 }
 
 // Windows Vista - 7
@@ -54,6 +50,7 @@ fn find_latest_windows_sdk_bin_dir(arch: Arch) -> Option<PathBuf> {
         .and_then(|reg_key| reg_key.get_value::<String, _>("CurrentInstallFolder"))
         .ok()
         .and_then(|root_dir| try_bin_dir(root_dir, "Bin", "Bin/x64", arch))
+        .and_then(try_rc_exe)
 }
 
 fn try_bin_dir(root_dir: String, x86_bin: &str, x64_bin: &str, arch: Arch) -> Option<PathBuf> {
@@ -64,3 +61,8 @@ fn try_bin_dir(root_dir: String, x86_bin: &str, x64_bin: &str, arch: Arch) -> Op
     }
     if p.is_dir() { Some(p) } else { None }
 }
+
+fn try_rc_exe(mut pb: PathBuf) -> Option<PathBuf> {
+    pb.push("rc.exe");
+    if pb.exists() { Some(pb) } else { None }
+}

```

**Comment by iliekturtles** [comment]
I think it looks good. Will try testing later today.

**Comment by iliekturtles** [comment]
Success: `Some("C:\\Program Files (x86)\\Windows Kits\\8.1\\bin/x64\\rc.exe")`

Sharing the same logic as `rustc`/`gcc` crates would be better, but this is a good short-term solution.

For anyone else coming here with the same issue you can work-around the problem by temporarily adding `rc.exe` to your path:

cmd:
```
> SET PATH=%PATH%;C:\Program Files (x86)\Windows Kits\8.1\bin\x64
> cargo ...
```

bash:
`> PATH=$PATH:/c/Program\ Files\ \(x86\)/Windows\ Kits/8.1/bin/x64 cargo ...`

**Comment by nabijaczleweli** [comment]
Released in [`v1.1.1`](https://github.com/nabijaczleweli/rust-embed-resource/releases/tag/v1.1.1)
